yente/routers/util.py

```diff
@@ -1,31 +1,48 @@
-from typing import Type
+from typing import List, Type
 from fastapi import Path, Query
 from fastapi import HTTPException
-from nomenklatura.matching import ALGORITHMS, ScoringAlgorithm, get_algorithm
+from nomenklatura.matching.logic_v2.model import LogicV2
+from nomenklatura.matching import (
+    ScoringAlgorithm,
+    LogicV1,
+    NameMatcher,
+    NameQualifiedMatcher,
+    RegressionV1,
+)
 
 from yente import settings
 from yente.data import get_catalog
 from yente.data.dataset import Dataset
 
+ENABLED_ALGORITHMS: List[Type[ScoringAlgorithm]] = [
+    LogicV1,
+    LogicV2,
+    NameMatcher,
+    NameQualifiedMatcher,
+    RegressionV1,
+]
 
 PATH_DATASET = Path(
     description="Data source or collection name to scope the query to.",
     examples=["default"],
 )
 QUERY_PREFIX = Query("", min_length=0, description="Search prefix")
 TS_PATTERN = r"^\d{4}-\d{2}-\d{2}(T\d{2}(:\d{2}(:\d{2})?)?)?$"
-ALGO_LIST = ", ".join([a.NAME for a in ALGORITHMS])
+ALGO_LIST = ", ".join([a.NAME for a in ENABLED_ALGORITHMS])
 ALGO_HELP = (
     f"Scoring algorithm to use, options: {ALGO_LIST} (best: {settings.BEST_ALGORITHM})"
 )
 
 
 def get_algorithm_by_name(name: str) -> Type[ScoringAlgorithm]:
     """Return the scoring algorithm class with the given name."""
-    name = name.lower().strip()
+    name = name.strip()
```

Comment:
this might be a breaking change if someone's sending in `LOGIC-V1` and it's been working until now, right?

is this because the new algo is called `UNSTABLE-logic-v2`?

An alternative is to lowercase a.NAME for the comparison

Reply:
Yeah, this is breaking, I realized this too (but only put it in the body of the commit message).

I think it's sort of weird that it's case insensitive, and figured that now's a good time to break it.

But also there is not really any good reason to break it, other than that it's a bit weird.

WDYT?

Reply:
I think stricter interfaces are better than loosey goosey in a lot of cases. But I don't think breaking the API for someone without an announcement is nice. I assumed this is a change we're planning on releasing in saas soon so people can start trying the algo if they know about it. A surprise breaking change doesn't really fit with what we're trying to do here. So I think somehow supporting case insensitive algo selection as we had before is kinda needed if we don't announce and give people a chance to adapt.